### ISSUE TITLE
xdpm package should fail with some error code when there are validation errors.

### ISSUE THREAD

**Issue by halfbloodman** [AdobeXD/xdpm#32]
### Expected Behaviour
`xdpm package` should fail with some error code when there are validation errors.

### Actual Behaviour
`xdpm package` shows validation errors but it returns "success" as execution code.

### Reproduce Scenario (including but not limited to)
I'm trying to configure plugin auto-build on Bitbucket pipelines. I put some commands including `yarn run xdpm package dist`.

Auto-build works fine but if (for example) the manifest is not correct - `package` command fails but Bitbucket shows that build pipeline worked without errors.

#### Sample Code that illustrates the problem

When I run `xdpm `from command line and if the manifest is incorrect - it gives following result:

```
yarn run xdpm package dist
yarn run v1.19.1
$ xdpm package dist
INFO: xdpm 3.0.2 - XD Plugin Manager CLI
INFO: Use of this tool means you agree to the Adobe Terms of Use at
https://www.adobe.com/legal/terms.html and the Developer Additional
Terms at https://wwwimages2.adobe.com/content/dam/acom/en/legal/servicetou/Adobe-Developer-Additional-Terms_en_US_20180605_2200.pdf.
ERROR: Can't package a plugin that has validation errors in the manifest.json:
.icons ({"limit":2}) -> should NOT have more than 2 items
.icons[2].width ({"allowedValues":[24,48]}) -> should be equal to one of the allowed values
.icons[2].height ({"allowedValues":[24,48]}) -> should be equal to one of the allowed values
Done in 0.49s.
```

It would be great if xdpm returned some error code if `package` command is failed. 
```
....
error Command failed with exit code 2.
```

In that case Bitbucket pipeline (and probably other CI tools) would be able to track

**Comment by ashryanbeats** [comment]
Thanks for reporting this @halfbloodman! This sounds like a reasonable request.

I'll leave it open for any further comment by others, including potential takers. PRs on this front are welcome! ⭐️

**Comment by pklaschka** [comment]
If no one else has done it by then, I'll take this the day after tomorrow :wink:. Since this is kind of CI-related, it kind of belongs to my "hobbyhorse" (I've found this translation for a German word called *"Steckenpferd"* referring to a personal favorite/enthusiastic project; no idea whether that translation works/is correct) in `xdpm`... :smile:

**Comment by ashryanbeats** [comment]
Nice! Looking forward to seeing what you come up with. 

***

Separately, we are very curious to learn more about how you're leveraging xdpm in your CI/auto-build workflows @pklaschka @halfbloodman.

**Comment by pklaschka** [comment]
> Separately, we are very curious to learn more about how you're leveraging xdpm in your CI/auto-build workflows @pklaschka @halfbloodman.

That's not too difficult, considering my plugin boilerplate is open-source (cf. https://github.com/pklaschka/xd-plugin-boilerplate) :wink: 

You can see the `.gitlab-ci.yml` [here](https://github.com/pklaschka/xd-plugin-boilerplate/blob/master/.gitlab-ci.yml). Basically, I trigger tests (unit tests first), and then, `npm run build` triggers `npm run compile && xdpm package dist`, where the `compile` step is just webpack doing its magic, compiling things into the `dist` folder, which then gets packaged into a `.xdx` which I can download from GitLab, meaning I have a `.xdx` file ready to download for every commit.

I'm also currently working on a semi-automated testing system for plugins (an individual software) where I'd be more then interested in integrating `xdpm`, although I haven't found a good way to do so, yet. That, however, is a whole different story and although somewhat related to CI/CD (which is why I'm mentioning it here), doesn't really belong into this issue :rofl:

**Comment by ashryanbeats** [comment]
@pklaschka did you end up taking this one on? If not, I might have a look and see if that can be rolled into 4.0 as well.

**Comment by pklaschka** [comment]
@ashryanbeats Feel free to give it a shot. I've done some research (stumbling onto `cli.error()`, `cli.debug()`, `process.exit(code)` etc.) and initial testing, but didn't get to anything that quite convinced me, yet. Since I'd probably only be able to continue my efforts on this after this week, it's pretty much still open :wink:

**Comment by ashryanbeats** [comment]
I'll push a branch that we can discuss.

**Comment by ashryanbeats** [comment]
@halfbloodman see the PR above and let me know if you have any feedback on that approach ⭐️

**Comment by ashryanbeats** [comment]
This should be addressed in v4.0, just published on npm:

```
npm i -g @adobe/xdpm
```
